sim-cli/src/config.rs

```diff
@@ -0,0 +1,241 @@
+use std::path::PathBuf;
+
+use config::{Config, File};
+use log::LevelFilter;
+use serde::Deserialize;
+
+use crate::Cli;
+
+/// Overwrites the fields on a SimulationConfig if corresponding arguments are supplied via the command line
+macro_rules! cli_overwrite {
+    ($config:expr, $cli:expr, $($field:ident),*) => {
+        match (&mut $config, $cli) {
+            (config_struct, cli_struct) => {
+                $(
+                    if let Some(value) = cli_struct.$field {
+                        log::info!(
+                            "SimulatinConfig::{} {:?} overwritten by CLI argument {:?}",
+                            stringify!($field),
+                            config_struct.$field,
+                            value
+                        );
+                        config_struct.$field = value;
+                    }
+                )*
+            }
+        }
+
+    };
+}
+
+#[derive(Debug, Deserialize)]
+pub struct SimulationConfig {
```

Comment:
Wondering whether we can move deserialization onto the existing cli struct and then just get CLI parsing to only override values that are present? 

Then we can create a simple `SimulationConfig` struct in `lib` can just convert the cli -> config struct like @okjodom suggested.